dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt

```diff
@@ -40,18 +44,29 @@ abstract class DataController<T>(
                 "for companyId '${companyAssociatedData.companyId}'",
         )
         val correlationId = generatedCorrelationId(companyAssociatedData.companyId)
-        val datasetToStore = buildDatasetToStore(companyAssociatedData, userId, uploadTime)
-
-        val dataIdOfPostedData = dataManager.addDataSet(datasetToStore, correlationId)
+        val dataIdOfPostedData = storeDataInStorage(companyAssociatedData, userId, uploadTime, correlationId)
         logger.info(
             "Posted company associated data for companyId '${companyAssociatedData.companyId}'. " +
                 "Correlation ID: $correlationId",
         )
         return ResponseEntity.ok(
-            DataMetaInformation(dataIdOfPostedData, dataType, userId, uploadTime, companyAssociatedData.companyId),
+            DataMetaInformation(
+                dataIdOfPostedData, dataType, userId, uploadTime,
+                companyAssociatedData.companyId, QAStatus.Pending,
+            ),
         )
     }
 
+    private fun storeDataInStorage(
```

Comment:
I think this method does not really help making the code more readable.

I think things get cleaner if this method is inlined.

Reply:
ok